File: 0406-queue-reconstruction-by-height/0406-queue-reconstruction-by-height.java
class Solution {
    public int[][] reconstructQueue(int[][] people) {
        Arrays.sort(people, (a1, a2) -> {
            if (a1[0] != a2[0]) {
                return a1[0] - a2[0];
            }
            return a2[1]-a1[1];
        });
        final List<int[]> res = new ArrayList<>();
        final int n = people.length;
        for (int i = n - 1; i >= 0; i--) {
            res.add(people[i][1], people[i]);
        }
        return res.toArray(new int[0][0]);
    }
}

File: 0406-queue-reconstruction-by-height/0406-queue-reconstruction-by-height.rs
impl Solution {
    pub fn reconstruct_queue(mut people: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        people.sort_by_key(|x| (-x[0], x[1]));
        let mut res = Vec::with_capacity(people.len());
        for person in people.iter() {
            res.insert(person[1] as usize, person.to_vec());
        }
        res
    }
}
